File: background-process/LatihanBroadcastReceiver/java/app/src/main/java/com/dicoding/picodiploma/mybroadcastreceiver/SmsReceiverActivity.java
package com.dicoding.picodiploma.mybroadcastreceiver;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class SmsReceiverActivity extends AppCompatActivity {

    public static final String EXTRA_SMS_NO = "extra_sms_no";
    public static final String EXTRA_SMS_MESSAGE = "extra_sms_message";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_sms_receiver);

        setTitle(getString(R.string.incoming_message));

        Button btnClose = findViewById(R.id.btn_close);
        TextView tvFrom = findViewById(R.id.tv_from);
        TextView tvMessage = findViewById(R.id.tv_message);

        btnClose.setOnClickListener(v -> finish());

        String senderNo = getIntent().getStringExtra(EXTRA_SMS_NO);
        String senderMessage = getIntent().getStringExtra(EXTRA_SMS_MESSAGE);

        tvFrom.setText(String.format("from : %s", senderNo));
        tvMessage.setText(senderMessage);
    }
}

File: background-process/LatihanBroadcastReceiver/kotlin/app/src/main/java/com/dicoding/picodiploma/mybroadcastreceiver/SmsReceiverActivity.kt
package com.dicoding.picodiploma.mybroadcastreceiver

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.dicoding.picodiploma.mybroadcastreceiver.databinding.ActivitySmsReceiverBinding

class SmsReceiverActivity : AppCompatActivity() {

    companion object {
        const val EXTRA_SMS_NO = "extra_sms_no"
        const val EXTRA_SMS_MESSAGE = "extra_sms_message"
    }

    private var binding: ActivitySmsReceiverBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivitySmsReceiverBinding.inflate(layoutInflater)
        setContentView(binding?.root)

        title = getString(R.string.incoming_message)

        binding?.btnClose?.setOnClickListener {
            finish()
        }

        val senderNo = intent.getStringExtra(EXTRA_SMS_NO)
        val senderMessage = intent.getStringExtra(EXTRA_SMS_MESSAGE)

        binding?.tvFrom?.text = getString(R.string.from, senderNo)
        binding?.tvMessage?.text = senderMessage
    }

    override fun onDestroy() {
        super.onDestroy()
        binding = null
    }
}
